cmd/cloudFoundryDeleteService.go

```
@@ -0,0 +1,66 @@
+package cmd
+
+import (
+	"github.com/SAP/jenkins-library/pkg/command"
+	"github.com/SAP/jenkins-library/pkg/log"
+)
+
+func cloudFoundryDeleteService(CloudFoundryDeleteServiceOptions cloudFoundryDeleteServiceOptions) error {
+
+	c := command.Command{}
+
+	// reroute command output to logging framework
+	c.Stdout(log.Entry().Writer())
+	c.Stderr(log.Entry().Writer())
+
+	cloudFoundryLogin(CloudFoundryDeleteServiceOptions, &c)
+	cloudFoundryDeleteServiceFunction(CloudFoundryDeleteServiceOptions.CfServiceInstance, &c)
+	cloudFoundryLogout(&c)
+
+	return nil
+}
+
+func cloudFoundryLogin(CloudFoundryDeleteServiceOptions cloudFoundryDeleteServiceOptions, c execRunner) error {
+	var cfLoginScript = []string{"login", "-a", CloudFoundryDeleteServiceOptions.CfAPIEndpoint, "-o", CloudFoundryDeleteServiceOptions.CfOrg, "-s", CloudFoundryDeleteServiceOptions.CfSpace, "-u", CloudFoundryDeleteServiceOptions.Username, "-p", CloudFoundryDeleteServiceOptions.Password}
+
+	log.Entry().WithField("cfAPI:", CloudFoundryDeleteServiceOptions.CfAPIEndpoint).WithField("cfOrg", CloudFoundryDeleteServiceOptions.CfOrg).WithField("space", CloudFoundryDeleteServiceOptions.CfSpace).WithField("password", CloudFoundryDeleteServiceOptions.Password).Info("Logging into Cloud Foundry..")
+
+	err := c.RunExecutable("cf", cfLoginScript...)
+	if err != nil {
+		log.Entry().
+			WithError(err).
+			Fatal("Failed to login to Cloud Foundry")
+	}
+	log.Entry().Info("Logged in successfully to Cloud Foundry..")
+	return err
+}
+
+func cloudFoundryDeleteServiceFunction(service string, c execRunner) error {
+	var cfdeleteServiceScript = []string{"delete-service", service, "-f"}
+
+	log.Entry().WithField("cfService", service).Info("Deleting the requested Service")
+
+	err := c.RunExecutable("cf", cfdeleteServiceScript...)
+	if err != nil {
```

please have a look if you can log out before failing.
Maybe return an error here and use `Fatal` only one level above.

---

If deletion fails, it now returns an error and using fatal one level above.
Should log out either way now, even if service deletion fails.